java/simulated_ble/README.md

```diff
@@ -0,0 +1,306 @@
+    // ...
+    /* Use publish() method to do the output. You are
+       allowed to publish your new Message instance. */
+    this.publish(message);
+  }
+}
+```
+
+### Simulated BLE Module
+
+| Input | Processor | Output                   | Source File               |
+| ----- | --------- | ------------------------ | ------------------------- |
+| N/A   | N/A       | Temperature data message | `SimulatedBleModule.java` |
+
+This module simulates a [BLE](https://en.wikipedia.org/wiki/Bluetooth_Low_Energy) temperature sensor. It uses its own [Timer](https://docs.oracle.com/javase/8/docs/api/java/util/Timer.html) to send randomized temperature data at a fixed rate. It also includes a MAC Address and source type in the message properties.
+
+Both the rate and the MAC address are coming from the module configuration string (typically it is a JSON string). That's why we need to first parse the configuration in the constructor of our module:
+
+```java
+public SimulatedBleModule(long address, Broker broker, String configuration) {
+  super(address, broker, configuration);
+  JSONObject bleConfig = new JSONObject(configuration);
+  this.macAddress = bleConfig.getString("macAddress");
+  this.messagePeriod = bleConfig.getInt("messagePeriod");
+}
+```
+
+When the module is started, a timer needs to be created to send temperature data. We do not need to release any resources (including the timer) because our timer is a [daemon](https://docs.oracle.com/javase/8/docs/api/java/lang/Thread.html#setDaemon-boolean-).
+
+```java
+@Override
+public void start() {
+  new Timer(true).schedule(new TimerTask() {
+    private Random rand = new Random();
+
+    @Override
+    public void run() {
+      try {
+        HashMap<String, String> properties = new HashMap<>();
+        properties.put("source", "bleTelemetry");
+        properties.put("macAddress", SimulatedBleModule.this.macAddress);
+
+        float temperature = this.rand.nextFloat() * 20 + 20;
+        byte[] content = Float.toString(temperature).getBytes();
+        
+        SimulatedBleModule.this.publish(new Message(content, properties));
+      } catch (Exception e) {
+        e.printStackTrace();
+      }
+    }
+  }, 0, this.messagePeriod);
+}
+```
+
+### Converter Module
+
+| Input                    | Processor                              | Output                 | Source File            |
+| ------------------------ | -------------------------------------- | ---------------------- | ---------------------- |
+| Temperature data message | Parse and construct a new JSON message | Structure JSON message | `ConverterModule.java` |
+
+This module is a typical Azure IoT Edge module. It accepts temperature message from another module (a hardware module, or in this case our simulated BLE module); and then it normalizes the temperature message to a structured JSON message (including appending the message ID, setting the property of whether we need to trigger the temperature alert, and so on).
```

Comment:
This module is a typical Azure IoT Edge module. It accepts temperature messages from other modules (a hardware module, or in this case our simulated BLE module); and then normalizes the temperature message in to a structured JSON message (including appending the message ID, setting the property of whether we need to trigger the temperature alert, and so on).